dio/lib/src/dio_mixin.dart

```diff
@@ -359,19 +361,16 @@ abstract class DioMixin implements Dio {
     ProgressCallback? onSendProgress,
     ProgressCallback? onReceiveProgress,
   }) async {
-    options ??= Options();
-    final requestOptions = options.compose(
+    final requestOptions = (options ??= Options()).compose(
       this.options,
       path,
       data: data,
       queryParameters: queryParameters,
       onReceiveProgress: onReceiveProgress,
       onSendProgress: onSendProgress,
       cancelToken: cancelToken,
+      sourceStackTrace: StackTrace.current,
     );
-    requestOptions.onReceiveProgress = onReceiveProgress;
-    requestOptions.onSendProgress = onSendProgress;
-    requestOptions.cancelToken = cancelToken;
```

Comment:
Are these unnecessary anymore? I didn't see new coding applying this.l

Reply:
They are already passed to `Options.compose` just above.